FreeRTOS_DNS.c

```diff
@@ -538,47 +520,99 @@
         return ulIPAddress;
     }
 
+/*!
+ * @brief prepare the buffer before sending
+ * @param [in] pcHostName
+ * @param [in] uxIdentifier identifier
+ * @param [in] xDNSSocket a valid socket
+ * @param [in] xAddress address structure
+ * @param [in] bHasDot hostname has dot
+ * @returns whether sending the data was successful
+ */
+    static BaseType_t prvSendBuffer( const char * pcHostName,
+                                     TickType_t uxIdentifier,
+                                     Socket_t xDNSSocket,
+                                     struct freertos_sockaddr * xAddress,
+                                     BaseType_t bHasDot )
+    {
+        BaseType_t xReturn = pdFAIL;
+        struct xDNSBuffer xDNSBuf = { 0 };
+        NetworkBufferDescriptor_t * pxNetworkBuffer = NULL;
+
+        /*get dns message buffer */
+        xDNSBuf.pucPayloadBuffer = prvGetPayloadBuffer( &pxNetworkBuffer,
+                                                        pcHostName );
+
+        if( xDNSBuf.pucPayloadBuffer != NULL )
+        {
+            xDNSBuf.uxPayloadLength = pxNetworkBuffer->xDataLength;
+            xDNSBuf.uxPayloadSize = pxNetworkBuffer->xDataLength;
+
+            #if ( ipconfigUSE_LLMNR == 1 )
+                if( bHasDot == pdFALSE )
+                {
+                    /* Use LLMNR addressing. */
+                    ( ipCAST_PTR_TO_TYPE_PTR( DNSMessage_t, xDNSBuf.pucPayloadBuffer ) )->usFlags = 0;
+                }
+            #else
+                ( void ) bHasDot;
+            #endif
+            xDNSBuf.uxPayloadLength = prvCreateDNSMessage( xDNSBuf.pucPayloadBuffer,
+                                                           pcHostName,
+                                                           uxIdentifier );
+
+            /* send the dns message */
+            xReturn = DNS_SendRequest( pcHostName,
+                                       uxIdentifier,
+                                       xDNSSocket,
+                                       xAddress,
+                                       &xDNSBuf );
+
+            if( xReturn == pdFAIL )
+            {
+                vReleaseNetworkBufferAndDescriptor( pxNetworkBuffer );
+            }
+        }
+
+        return xReturn;
+    }
+
 /*!
  * @brief main dns operation description function
  *
  * @param [in] pcHostName hostname to get its ip address
- * @param [in] uxIdentifier
+ * @param [in] uxIdentifier identifier
```

Comment:
I am not sure that I understand what the description identifier means? What ID is this?